packages/starknet_provider/lib/src/model/block_with_receipts.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:starknet/starknet.dart';
+import 'package:starknet_provider/src/model/get_transaction_receipt.dart';
+
+part 'block_with_receipts.freezed.dart';
+part 'block_with_receipts.g.dart';
+
+@freezed
+class TransactionWithReceipt with _$TransactionWithReceipt {
+  const factory TransactionWithReceipt({
+    @JsonKey(name: 'transaction_hash') required Felt transactionHash,
+    required String type,
+    required TxnReceipt receipt,
+  }) = _TransactionWithReceipt;
+
+  factory TransactionWithReceipt.fromJson(Map<String, dynamic> json) =>
+      _$TransactionWithReceiptFromJson(json);
+}
+
+@freezed
+class BlockWithReceipts with _$BlockWithReceipts {
+  const factory BlockWithReceipts({
+    @JsonKey(name: 'block_hash') String? blockHash,
+    @JsonKey(name: 'parent_hash') String? parentHash,
+    @JsonKey(name: 'block_number') int? blockNumber,
+    @JsonKey(name: 'sequencer_address') String? sequencerAddress,
+    @JsonKey(name: 'new_root') String? newRoot,
+    int? timestamp,
```

Comment:
Why using `String` instead of `Felt` ?
In JSON RPC it's specified as `FELT`:
```
            "BLOCK_HASH": {
                "title": "Block hash",
                "$ref": "#/components/schemas/FELT"
            },
```